LibGAP.jl/src/julia_to_gap.jl

```diff
@@ -0,0 +1,99 @@
+## Converters
+
+## Default
+julia_to_gap(x::GAPInputType, ::Val{Recursive}=Val(false), recursion_dict = nothing) where Recursive = x
```

Comment:
Please add a longer comment to this file which explains what is going on. Like: What is this `Val{Recursive}` bit doing; but also the general idea of what kind of methods, with what signature, one has to achieve what.

I actually need that explanation, as I don't understand right now why this complicated setup is better than this simpler one: For all "basic" types (= not containing subobjects that need to be converted), have methods like this:
```julia
julia_to_gap(x::Int32, recursion_dict) = Int64(x)
```
Then, have one generic redispatch function which handles all these primitive ones at once; and for all data types with subobjects, have both 1 and 2 arg methods:
```julia
julia_to_gap(x::Any) = julia_to_gap(x, nothing)

function julia_to_gap(obj::Array{T,1}, recursion_dict = IdDict()) where T
...
end
```

Perhaps the code the above produces is suboptimal? Well, I wouldn't ask this question if there was a comment explaining the design of the current approach :-)

Reply:
I am not sure about the code being less optimal that way. I will try to implement the changes you propose.